Makefile.include

```diff
@@ -205,12 +205,6 @@ include $(RIOTMAKE)/boards.inc.mk
 include $(RIOTMAKE)/dependencies_debug.inc.mk
 
 # Use TOOLCHAIN environment variable to select the toolchain to use.
-# If native, TOOLCHAIN for OSX is llvm
-ifeq ($(BOARD),native)
-  ifeq ($(OS),Darwin)
-    TOOLCHAIN ?= llvm
-  endif
-endif
```

Comment:
Maybe keep this, but add an error message instead of setting the toolchain instead to mitigate some confused forum questions?

```mk
ifeq ($(BOARD),native)
  ifeq ($(OS),Darwin)
    $(shell $(COLOR_ECHO) "$(COLOR_RED)"Buildin on macOS is not supported."\
	    "We recommend vagrant for building:$(COLOR_RESET)"\
	    "https://github.com/RIOT-OS/RIOT/blob/master/dist/tools/vagrant/README.md 1>&2)
  endif
endif

Reply:
I implemented your suggestion. Still builds on my linux machine. On macOS, the error is printed but it still tries to build the application (which then fails ofc).